crates/uv-distribution/src/source/mod.rs

```diff
@@ -1137,7 +1156,10 @@ impl<'a, T: BuildContext> SourceDistributionBuilder<'a, T> {
         {
             if let Some(metadata) = read_cached_metadata(&metadata_entry).await? {
                 debug!("Using cached metadata for: {source}");
-                return Ok(ArchiveMetadata::from(metadata));
+                return Ok(ArchiveMetadataLowered::from(
+                    Metadata23Lowered::from_tool_uv(metadata, fetch.path(), self.preview_mode)
```

Comment:
@konstin - Do you think we should be caching the lowered or non-lowered metadata here? What are the implications of caching the non-lowered metadata, then lowering it after?